### PR TITLE
AP_BattMonitor: smbus remove unused returns on reads

### PR DESCRIPTION
**Pull request by hendjoshsr71**
Removes the unused bool returns on the SMBus read register functions. 
Removes an unneeded temporary variable in `read_full_charge_capacity`.

Saves 64-72B depending on vehicle

### CODE REVIEW

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp [hunk 1]
```diff
@@ -46,71 +46,58 @@ void AP_BattMonitor_SMBus::read(void)
...
 {
     uint16_t data;
-    if (read_word(BATTMONITOR_SMBUS_TEMP, data)) {
-        _has_temperature = (AP_HAL::millis() - _state.temperature_time) <= AP_BATT_MONITOR_TIMEOUT;
-
-        _state.temperature_time = AP_HAL::millis();
-        _state.temperature = ((float)(data - 2731)) * 0.1f;
-        return true;
+    if (!read_word(BATTMONITOR_SMBUS_TEMP, data)) {
+        _has_temperature = false;
```
Comment (khancyr) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp:80]: miss a return here

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp [hunk 2]
```diff
@@ -46,71 +46,58 @@ void AP_BattMonitor_SMBus::read(void)
...
-
     if (_full_charge_capacity != 0) {
-        return true;
-    } else if (read_word(BATTMONITOR_SMBUS_FULL_CHARGE_CAPACITY, data)) {
-        _full_charge_capacity = data * get_capacity_scaler();
-        return true;
+        return;
     }
-    return false;
+    read_word(BATTMONITOR_SMBUS_FULL_CHARGE_CAPACITY, _full_charge_capacity);
```
Comment (khancyr) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp:55]: let's be pedantic here and keep checking the read_word return, that is cheaper than the *= anyway

**Review by khancyr — CHANGES_REQUESTED** [review]
Not really NFC as some logic were moved.
We truely don't use the return on those functions and it doesn't seems very usefull to have since most function are re-entrant and does correct checking.

**Review by WickedShell — CHANGES_REQUESTED** [review]
The temperature reading stuff is a bit of a preexsisting bug which your PR highlights, but it only slightly improves the failure case, and in some ways makes it worse. I think the suggestions will fix it. I'm good with the changes after the handling of `_has_temperature` gets looked at more.

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp [hunk 3]
```diff
@@ -46,71 +46,61 @@ void AP_BattMonitor_SMBus::read(void)
...
+        _has_temperature = false;
+        return;
     }
-    
-    _has_temperature = false;
+    _has_temperature = (AP_HAL::millis() - _state.temperature_time) <= AP_BATT_MONITOR_TIMEOUT;
 
-    return false;
+    _state.temperature_time = AP_HAL::millis();
+    _state.temperature = ((float)(data - 2731)) * 0.1f;
```
[the remark this answers is not in the document]
Comment (hendjoshsr71) [reply libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: Done.

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp [hunk 4]
```diff
@@ -46,71 +46,61 @@ void AP_BattMonitor_SMBus::read(void)
...
-        _state.temperature_time = AP_HAL::millis();
-        _state.temperature = ((float)(data - 2731)) * 0.1f;
-        return true;
+    if (!read_word(BATTMONITOR_SMBUS_TEMP, data)) {
+        _has_temperature = false;
+        return;
     }
-    
-    _has_temperature = false;
+    _has_temperature = (AP_HAL::millis() - _state.temperature_time) <= AP_BATT_MONITOR_TIMEOUT;
```
[the remark this answers is not in the document]
Comment (hendjoshsr71) [reply libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: Done.

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp [hunk 5]
```diff
@@ -46,71 +46,61 @@ void AP_BattMonitor_SMBus::read(void)
...
     uint16_t data;
-    if (read_word(BATTMONITOR_SMBUS_TEMP, data)) {
-        _has_temperature = (AP_HAL::millis() - _state.temperature_time) <= AP_BATT_MONITOR_TIMEOUT;
-
-        _state.temperature_time = AP_HAL::millis();
-        _state.temperature = ((float)(data - 2731)) * 0.1f;
-        return true;
+    if (!read_word(BATTMONITOR_SMBUS_TEMP, data)) {
+        _has_temperature = false;
+        return;
```
[the remark this answers is not in the document]
Comment (hendjoshsr71) [reply libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: Fixed. I should have been thinking a step beyond to what the full intent of the code was even if the conversion I did was ok.

**Comment by hendjoshsr71** [comment]
Addressed the changes requested above.

**Review by WickedShell — APPROVED** [review]
> I should have been thinking a step beyond to what the full intent of the code was even if the conversion I did was ok.

No worries, it's one of those things that was much more apparent in the review, and when you are just refactoring it's easier to miss. Thanks for making the changes.

**Review by peterbarker — APPROVED** [review]
These should probably be `update_temp_reading` now.  But meh.

**Pull request merged**